Java-FlightServer/src/main/java/ibm/com/example/ExampleFlightServer.java

```diff
@@ -0,0 +1,156 @@
+        this.isNonBlocking = isNonBlocking;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public void start() throws IOException {
+        flightServer.start();
+    }
+
+    public void awaitTermination() throws InterruptedException {
+        flightServer.awaitTermination();
+    }
+
+    /**
+     *  Main method starts the server listening to localhost:12233.
+     */
+    public static void main(String[] args) throws Exception {
+        final BufferAllocator a = new RootAllocator(Long.MAX_VALUE);
+        final ExampleFlightServer efs = new ExampleFlightServer(a, Location.forGrpcInsecure("localhost", 12233));
+        efs.start();
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                System.out.println("\nExiting...");
+                AutoCloseables.close(efs, a);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }));
+        efs.awaitTermination();
+    }
+
+    private final class Producer extends NoOpFlightProducer {
+        private final int RecordsPerBatch = 1024*1024;
+        private final VectorSchemaRoot constVectorSchemaRoot = getConstVectorSchemaRoot();
+
+        private VectorSchemaRoot getConstVectorSchemaRoot() {
+            final Schema schema = new Schema(ImmutableList.of(
+                    Field.nullable("a", MinorType.BIGINT.getType()),
+                    Field.nullable("b", MinorType.BIGINT.getType()),
+                    Field.nullable("c", MinorType.BIGINT.getType()),
+                    Field.nullable("d", MinorType.BIGINT.getType())
+            ));
+            //Schema schema = Schema.deserialize(ByteBuffer.wrap(perf.getSchema().toByteArray()));
+            VectorSchemaRoot root = VectorSchemaRoot.create(schema, allocator);
+            //root.allocateNew();
+            BigIntVector a = (BigIntVector) root.getVector("a");
+            BigIntVector b = (BigIntVector) root.getVector("b");
+            BigIntVector c = (BigIntVector) root.getVector("c");
+            BigIntVector d = (BigIntVector) root.getVector("d");
+
+            for (int j = 0; j < this.RecordsPerBatch; j++) {
+                a.setSafe(j, j);
+                b.setSafe(j, j);
+                c.setSafe(j, j);
+                d.setSafe(j, j);
+            }
+            root.setRowCount(this.RecordsPerBatch);
+            return root;
+        }
+
+        @Override
+        public void getStream(CallContext context, Ticket ticket,
+                              ServerStreamListener listener) {
+            final Runnable loadData = () -> {
+                listener.setUseZeroCopy(true);
+                listener.start(this.constVectorSchemaRoot);
+                for (int i=0; i<1000; i++) {
+                    listener.putNext();
+                }
+                listener.completed();
+            };
+
+            if (!isNonBlocking) {
```

Comment:
Can you explain this pattern? Is flight in Java designed such that `getStream` can't be called in parallel unless we execute the logic in a thread on our own? 

Also, which one are we using currently (blocking or not)

Reply:
@roee88 : we run with `isNonBlocking=false` (in other words, we are blocking).
Java can work in parallel for different tickets. This piece of code refers to handling a single ticket.